src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -79,5 +83,197 @@ public string VersionString
                 return _versionString;
             }
         }
+
+        /// <summary>
+        /// Indicates whether the current application is running on the specified platform.
+        /// </summary>
+        /// <param name="platform">Case-insensitive platform name. Examples: Browser, Linux, FreeBSD, Android, iOS, macOS, tvOS, watchOS, Windows.</param>
+        public static bool IsOSPlatform(string platform)
+        {
+            if (platform == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.platform);
+            }
+            if (platform.Length == 0)
```

Comment:
Is it really worth it to explicitly check and throw for an empty string here?

Reply:
another good point, I've removed this check and we are now simply going to return false